Orange/data/table.py

```diff
@@ -1535,7 +1535,8 @@ def has_missing_class(self):
     def get_nan_frequency_attribute(self):
         if self.X.size == 0:
             return 0
-        return np.isnan(self.X).sum() / self.X.size
+        vals = self.X if not sp.issparse(self.X) else self.X.data
+        return np.isnan(vals).sum() / self.X.size
```

Comment:
```
>>> sp.csc_matrix([[1, 2, 0], [0, 0, 1]]).size
3
```

This code assumes it's 6, right?

Reply:
I thought I had an answer, but looking at it again the concept of nan frequency in sparse arrays bugs me. What is the number we actually care about here?

Reply:
In sparse matrices, the absent values are zeros, not missing. Hence the correct denominator is `self.X.shape[0] * self.X.shape[1]`, I think.

Reply:
I think that instead of the size, as Janez pointed out, we should use the thing that returns "6". :)

Orange does not count NaNs consistently. The code by Niko usually returns number of zero elements. I'd prefer it to return the number of actual "NaN" values. So I'd go with the actual number of NaNs.

Reply:
As far as I checked - and I did check -, there seems to be no thing that returns 6.

Reply:
`vals.size` doesn't return 6?